packages/connection-form/src/utils/csfle-handler.ts

```diff
@@ -184,10 +206,55 @@ export function handleUpdateCsfleKmsTlsParam({
 // as an option, regardless of whether it is filled. Consequently, we need
 // to set it to undefined explicitly if the user wants to disable automatic
 // CSFLE entirely (indicated by removing all CSFLE options).
-export function unsetAutoEncryptionIfEmpty(
-  o?: AutoEncryptionOptions
-): AutoEncryptionOptions | undefined {
-  return o && hasAnyCsfleOption(o) ? o : undefined;
+export function unsetFleOptionsIfEmptyAutoEncryption(
+  connectionOptions: Readonly<ConnectionOptions>
+): ConnectionOptions {
+  connectionOptions = cloneDeep(connectionOptions);
+  const autoEncryption =
+    connectionOptions.fleOptions?.autoEncryption &&
+    hasAnyCsfleOption(connectionOptions.fleOptions?.autoEncryption)
+      ? connectionOptions.fleOptions?.autoEncryption
+      : undefined;
+
+  if (!autoEncryption) {
+    return {
+      ...connectionOptions,
+      fleOptions: undefined,
+    };
+  }
+
+  function filterEmptyValues(obj: Record<string, any> | undefined) {
+    const values = Object.fromEntries(
+      Object.entries(obj ?? {})
+        .filter(([, v]) => Object.keys(v ?? {}).length > 0)
+        .map(([k, v]) => [k, v])
+    );
+    return Object.keys(values).length > 0 ? values : undefined;
```

Comment:
Nit, we can use generics here to improve type safety:

```suggestion
  function filterEmptyValues<T extends object>(obj: T | undefined): Partial<T> | undefined {
    const values = Object.fromEntries(
      Object.entries(obj ?? {})
        .filter(([, v]) => Object.keys(v ?? {}).length > 0)
    );
    return Object.keys(values).length > 0 ? values as Partial<T> : undefined;
```